pkg/util/testing/wrappers.go

```diff
@@ -143,6 +151,15 @@ func (w *WorkloadWrapper) Condition(condition metav1.Condition) *WorkloadWrapper
 	return w
 }
 
+func (w *WorkloadWrapper) SetOrReplaceCondition(condition metav1.Condition) *WorkloadWrapper {
+	existingCondition := apimeta.FindStatusCondition(w.Status.Conditions, condition.Type)
+	if existingCondition != nil {
+		apimeta.RemoveStatusCondition(&w.Status.Conditions, condition.Type)
+	}
+	apimeta.SetStatusCondition(&w.Status.Conditions, condition)
+	return w
+}
```

Comment:
Probably, we don't need this wrapper since the `apimeta.SetStatusCondition` checks whether the conditions already have the condition.

https://github.com/kubernetes/apimachinery/blob/61b874e0403b27353daf90f9789c762610731ab6/pkg/api/meta/conditions.go#L25-L55

Reply:
I use it here https://github.com/kubernetes-sigs/kueue/pull/717/files#diff-2983f8941450021febb0b5013e359a5f344a84a6f6a9cfee8ecbdf3c3f92a9cfR756

I need to replace LastTransitionTime from condition but SetOrReplaceCondition doesn't do that if !newCondition.LastTransitionTime.IsZero().

Reply:
Ah, you want to update the LastTransitionTime if the oldCondition.Status equals the newCondition.Status, right?
If so, sgtm.